src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs

```diff
@@ -97,7 +104,7 @@ static int Main(string[] args)
         {
             Console.WriteLine("[XUnitLogChecker]: An error occurred. No stats csv"
                             + $" was found. The expected name would be '{statsCsvPath}'.");
-            return FAILURE;
+            return TestExitCode;
```

Comment:
Is it possible for the test to return success, but the XUnitLogChecker to fail (i.e., places where it previously returned `FAILURE` before this PR)? In that case, should the CI consider that a failure? Or do you want various XUnitLogChecker failures to be ignored by the CI? If you want XUnitLogChecker failures to be shown as a CI failure, then the `return FAILURE` cases here should be preserved.

Reply:
Thanks Bruce, that's a reasonable suggestion; in practice I think most of these special cases only occur when the test crashes halfway through and then it also returns an error exit code but I believe your suggestion is a general goodness for robustness of the infra.

Reply:
This sounds like a reasonable suggestion to me, combined with Juan's in the next comment. I'll implement it.